Compilation failure with GCC 7.5.0

**Describe the bug**
QMCPACK fails to build when GCC 7.5.0 is used (released Nov. 14, 2019).  The issue relates to private access:

```
[ 20%] Building CXX object src/QMCWaveFunctions/CMakeFiles/qmcwfs.dir/BsplineFactory/createComplexDouble.cpp.o
In file included from /home/j1k/projects/ecp/16_batched_nofk/qmcpack_batched_nofk/src/QMCWaveFunctions/BsplineFactory/createComplexDouble.cpp:30:0:
/home/j1k/projects/ecp/16_batched_nofk/qmcpack_batched_nofk/src/QMCWaveFunctions/BsplineFactory/HybridRepSetReader.h: In instantiation of ‘void qmcplusplus::HybridRepSetReader<SA>::initialize_hybridrep_atomic_centers() [with SA = qmcplusplus::HybridRepCplx<qmcplusplus::SplineC2R<double> >]’:
/home/j1k/projects/ecp/16_batched_nofk/qmcpack_batched_nofk/src/QMCWaveFunctions/BsplineFactory/createComplexDouble.cpp:103:1:   required from here
/home/j1k/projects/ecp/16_batched_nofk/qmcpack_batched_nofk/src/QMCWaveFunctions/BsplineFactory/HybridRepSetReader.h:165:16: error: ‘qmcplusplus::HybridRepCenterOrbitals<double>::smoothing_schemes qmcplusplus::HybridRepCenterOrbitals<double>::smooth_scheme’ is private within this context
       bspline->smooth_scheme = SA::smoothing_schemes::CONSISTENT;
       ~~~~~~~~~^~~~~~~~~~~~~
```

**To Reproduce**
Steps to reproduce the behavior:
1. release version or git commit hash being built

```
commit 6161091baa344f06eb618ed7c859a287427a7630 (HEAD -> develop, main/develop)
Merge: 9818a4703 897da66e1
Author: Ye Luo <yeluo@anl.gov>
Date:   Mon Aug 30 09:32:46 2021 -0500

    Merge pull request #3257 from shivupa/spinonebody
    
    One Body Spin dependent Jastrow
```

2. cmake command

On Ubuntu 18.04.5 LTS:

```
export CXX=mpiCC

echo "building soa real"
mkdir build_soa_real
cd build_soa_real
cmake -DENABLE_SOA=1 ..
make -j 24
cd ..
ln -s ./build_soa_real/bin/qmcpack qmcpack_soa_real
```


**Expected behavior**
Clean build.

**System:**
 - system name: Ubuntu laptop

--- a/src/QMCWaveFunctions/BsplineFactory/HybridRepCenterOrbitals.h
+++ b/src/QMCWaveFunctions/BsplineFactory/HybridRepCenterOrbitals.h
@@ -406,7 +406,11 @@ public:
   using RealType     = typename DistanceTableData::RealType;
   using PosType      = typename DistanceTableData::PosType;
 
+#if (defined(__GNUC__) && (__GNUC__== 7))
+// Workaround GCC 7.5 and presumed earlier bug. Leave member data public.
+#else
 private:
+#endif
   ///atomic centers
   std::vector<AtomicOrbitals<ST>> AtomicCenters;
   ///table index

